code/game/objects/items/stacks/medical.dm

```diff
@@ -149,3 +163,82 @@
 /obj/item/stack/medical/ointment/suicide_act(mob/living/user)
 	user.visible_message("<span class='suicide'>[user] is squeezing \the [src] into [user.p_their()] mouth! [user.p_do(TRUE)]n't [user.p_they()] know that stuff is toxic?</span>")
 	return TOXLOSS
+
+/obj/item/stack/medical/suture
+	name = "suture"
+	desc = "Sterile sutures used to seal up cuts and lacerations."
+	gender = PLURAL
+	singular_name = "suture"
+	icon_state = "suture"
+	self_delay = 30
+	other_delay = 10
+	amount = 15
+	max_amount = 15
+	repeating = TRUE
+	var/heal_brute = 10
+	grind_results = list(/datum/reagent/medicine/spaceacillin = 2)
+
+/obj/item/stack/medical/suture/medicated
+	name = "medicated suture"
+	icon_state = "suture_purp"
+	desc = "A suture infused with drugs that speed up wound healing of the treated laceration."
+	heal_brute = 15
+	grind_results = list(/datum/reagent/medicine/polypyr = 2)
+
+/obj/item/stack/medical/suture/heal(mob/living/M, mob/user)
+	. = ..()
+	if(M.stat == DEAD)
+		to_chat(user, "<span class='warning'>[M] is dead! You can not help [M.p_them()].</span>")
+		return
+	if(iscarbon(M))
+		return heal_carbon(M, user, heal_brute, 0)
+	to_chat(user, "<span class='warning'>You can't heal [M] with the \the [src]!</span>")
+
+/obj/item/stack/medical/mesh
+	name = "regenerative mesh"
+	desc = "A bacteriostatic mesh used to dress burns."
+	gender = PLURAL
+	singular_name = "regenerative mesh"
+	icon_state = "regen_mesh"
+	self_delay = 30
+	other_delay = 10
+	amount = 15
+	max_amount = 15
+	repeating = TRUE
+	var/heal_burn = 10
+	var/is_open = FALSE ///This var determines if the sterile packaging of the mesh has been opened.
+
+/obj/item/stack/medical/mesh/update_icon()
+	if(!is_open)
+		icon_state = "regen_mesh_closed"
+		return
+	. = ..()

```

Comment:
This proc seems to be missing the other possibility.

Reply:
The other possibility is to run parent update_icon to update the sprite according to stack size, I didn't come up with a better way of doing it.

Reply:
No. I meant that it won't display the correct icon_state if it's open.
```
	. = ..()
	if(is_open)
		icon_state = "regen_mesh"
	else
		icon_state = "regen_mesh_closed"
```
default icon state should also probably be regen_mesh_closed

Reply:
default can't be regen_mesh_closed.

```
/obj/item/stack/update_icon()
	if(novariants)
		return ..()
	if(amount <= (max_amount * (1/3)))
		icon_state = initial(icon_state)
	else if (amount <= (max_amount * (2/3)))
		icon_state = "[initial(icon_state)]_2"
	else
		icon_state = "[initial(icon_state)]_3"
	..()
```

Reply:
oh. I forgot I wrote that.

Reply:
I don't claim my implementation is elegant, but it seems to work in my testing so far.